### DISCUSSION TITLE
Artisan mode for hiding comments in newly created files

### DISCUSSION THREAD

**Discussion by johangit** [laravel/framework#52296] [Ideas]
The `artisan make` command is the best helper for creating new app files. However, the first thing I do after executing the CLI is delete the comments (`php artisan make:policy FormSettingsPolicy --model=FormSettings` → 67 lines file → comments removing → 37 lines file). I've been working with Laravel for a long time, and I understand the meaning of each function. For me, those comments are just additional lines that shift the focus away from the main content. Most parts of the framework already have understandable and readable function names and properties. So, the best solution for me is to remove the comments.

On the other hand, new developers really need these comments.

My suggestion is to add an option that can remove code comments in the result of the 'artisan make' command. I'm not sure if it's possible, but I just wanted to discuss it.

**Comment by lrljoe** [comment]
You can just run
```
php artisan stub:publish
```
This will publish the stub files to your project root in "stubs".   You can then customise the stubs as you wish, (e.g. removing the comments)

It depends on the maturity of your org/clients, adding comments is really best practice.  If you don't want them in production, then just strip the comments out as part of your build/deployment process.
